Material.Blazor/Components/MenuSurface/MBMenuSurface.razor.cs

```diff
@@ -0,0 +1,113 @@
+using Material.Blazor.Internal;
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+using System;
+using System.Threading.Tasks;
+
+namespace Material.Blazor
+{
+    /// <summary>
+    /// This is a general purpose Material Theme menu.
+    /// </summary>
+    public partial class MBMenuSurface : ComponentFoundation
+    {
+        /// <summary>
+        /// A render fragement as a set of <see cref="MBListItem"/>s.
+        /// </summary>
+        [Parameter] public RenderFragment ChildContent { get; set; }
+
+
+        /// <summary>
+        /// Regular, fullwidth or fixed positioning/width.
+        /// </summary>
+        [Parameter] public MBMenuSurfacePositioning MenuSurfacePositioning { get; set; } = MBMenuSurfacePositioning.Regular;
+
+
+        private DotNetObjectReference<MBMenuSurface> ObjectReference { get; set; }
+        private ElementReference ElementReference { get; set; }
+        private bool IsOpen { get; set; } = false;
+
+
+        // Would like to use <inheritdoc/> however DocFX cannot resolve to references outside Material.Blazor
+        protected override async Task OnInitializedAsync()
+        {
+            await base.OnInitializedAsync();
+
+            ClassMapperInstance
+                .Add("mdc-menu-surface mdc-menu-surface--fixed")
+                .AddIf(GetMenuSurfacePositioningClass(MenuSurfacePositioning), () => MenuSurfacePositioning != MBMenuSurfacePositioning.Regular);
+
+            ObjectReference = DotNetObjectReference.Create(this);
+        }
+
+
+        private bool _disposed = false;
+        protected override void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                ObjectReference?.Dispose();
+            }
+
+            _disposed = true;
+
+            base.Dispose(disposing);
+        }
+
+
+        /// <summary>
+        /// For Material Theme to notify of menu closure via JS Interop.
+        /// </summary>
+        [JSInvokable("NotifyClosedAsync")]
+        //[System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0060:Remove unused parameter", Justification = "<Pending>")]
+        public async Task NotifyClosedAsync()
+        {
+            IsOpen = false;
+            await Task.CompletedTask;
+        }
```

Comment:
Consider replacing with
```cs
[JSInvokable]
public void NotifyClosed()
{
   IsOpen = false;
}
```
I don't see a reason this call needs to be async at all. I also don't think the argument to JSInvokable is necessary unless the method name differs.